### ISSUE TITLE
Check viability of using Luca Damasco proof of concept with pyodide

### ISSUE THREAD

**Issue by berinhard** [berinhard/pyp5js#73]
Recently, @villares sent me [a tweet](https://twitter.com/Lucapodular/status/1137731900863389696?s=19) from @luxapodular showing a proof of concept integrating p5.js with Python using Pyodide.

I really think we could use the strategy he proposes in his proof of concept in `pyp5js` to remove `transcrypt` depedency. This would be a **huge deal** because [many issues](https://github.com/berinhard/pyp5js/labels/Transcrypt%27s%20Requirement) depending on `transcrypt` new features would be fixed.

Repo link: https://github.com/Luxapodular/Py5.js

**Comment by jeremydouglass** [comment]
https://luxapodular.github.io/Py5.js/ looks great -- especially given the many things that transcrypt can't import.

I think the demo is currently limited to 100x100 because size() isn't implemented so cutting a pasting some examples gives misleading results--but it is a great demo if you are aware of that limitation.

**Comment by jtpio** [comment]
This looks very promising.

It would be great to build on top of this indeed, so `pyp5js` can be reused in various environments (as an example [ipyp5](https://github.com/jtpio/ipyp5) is an attempt at making p5 work in Jupyter notebooks with Jupyter widgets)

**Comment by villares** [comment]
@jeremydouglass try `createCanvas()` (from p5js.org) instead of `size()`, I think it's working!

```python
def setup():
    createCanvas(400, 400)

def draw():
    rect(350, 350, 40, 40)
```

**Comment by berinhard** [comment]
Documenting a few important links:

- List of [pyodide releases](https://github.com/iodide-project/pyodide/releases) so we don't need to build it manually;
- @Luxapodular [wrapper](https://github.com/Luxapodular/Py5.js/blob/master/py5.js#L60) around p5.js with pyodide;
- Docs on [running pyodide from JS code](https://pyodide.readthedocs.io/en/latest/using_pyodide_from_javascript.html);
- [Javascript from Python](https://pyodide.readthedocs.io/en/latest/type_conversions.html?highlight=javascript#javascript-from-python) docs;
- pyodide 0.15.0 [CDN](https://pyodide-cdn2.iodide.io/v0.15.0/full/pyodide.js);

**Comment by berinhard** [comment]
Hey everyone, I was able to implement a proof of concept and I fully documented it on #123 PR (sorry for any eventual english grammar error).

I've also created the issue #124 to be the start of the final implementation of pyodide + pyp5js. I'd be glad to help or pair with anyone interested on helping with this =)

**Comment by berinhard** [comment]
I know there are some people in this issue interested in this discussion, so I invite you to take a loot in this demo I've released today:

https://berinhard.github.io/pyp5js/pyodide/

**Comment by jeremydouglass** [comment]
@berinhard -- sorry for I'm just seeing this now. This is extremely exciting!

Can you explain a bit how pyodide is being used here, and what the capabilities are? and/or prepopulate the demo window with a sample sketch that shows using the pyodide? I was able to draw an ellipse, but when for example I tried running:

```import numpy as np
```

It produced the browser console error:

> pyodide.asm.js:8 ModuleNotFoundError: No module named 'numpy'

**Comment by berinhard** [comment]
Hi @jeremydouglass sorry for taking to long to get back to this, but I was focused on working to properly integrate pyp5js with Pyodide. Now the project support multiple interpreters, pyodide (the new default one) and transcrypt. 

The pyodide's curent demo is really just an improvement on Luca's proof of concept so I could discover how big the impact of integrating pyp5js with it would be. Now the project implements pyodide integration and this opened room to improvements. To address your question, specifically, I created [this milestone](https://github.com/berinhard/pyp5js/milestone/4) with all new issues addressing my next goals on pyodide integration. There you can find the known limitations the mode has for now and keep track of the progress.

About the `numpy` question, currently it's not enabled because we're using a king of "lightweight" version of Pyodide, I guess. But this is addressed in #148 and should be next thing to work on when the milestone work is done.

**Comment by berinhard** [comment]
Feel free to try the [new 0.5.1 version](https://pypi.org/project/pyp5js/) out:

```
$ pip install pyp5js
$ pyp5js serve
```

And open http://127.0.0.1:5000/ to have fun =}
